obs-studio-server/source/nodeobs_api.cpp

```diff
@@ -814,9 +821,12 @@ typedef std::basic_string<char, ci_char_traits> istring;
 
 /* This should be reusable outside of node-obs, especially
 * if we go a server/client route. */
-void OBS_API::openAllModules(void)
+bool OBS_API::openAllModules(std::string& err)
 {
-	OBS_service::resetVideoContext(NULL);
+	if (!OBS_service::resetVideoContext(NULL)) {
```

Comment:
Do you think more checks would be enough or maybe a dedicated class to handle obs failures and format them to the frontend?

Reply:
See this as to what it could possibly return: https://github.com/obsproject/obs-studio/blob/047e586eafa43b139a4a605985592be75570fd61/libobs/obs.h#L351

Our API is a little unorthodox here unfortunately so it's a bit confusing, as far as design goes, where the check should properly go. 

If possible, we should just pass the value back to the frontend and let them handle it. This way they can emit a message or alert box to the user.